app/graphql/mutation_root.rb

```diff
@@ -13,21 +13,7 @@
     argument :subgroup, types.String
     argument :userId, types.Int
 
-    resolve ->(obj, args, ctx) {
-      workflow = Workflow.accessible_by(ctx[:credential]).find(args[:workflowId])
-      data_request = workflow.data_requests.build(
-        user_id: args[:userId],
-        subgroup: args[:subgroup],
-        requested_data: args[:requestedData]
-      )
-
-      data_request.status = DataRequest.statuses[:pending]
-      data_request.url = nil
-      data_request.save!
-
-      DataRequestWorker.perform_async(data_request.workflow_id)
-      data_request
-    }
+    resolve CreatesDataRequests.graphql
```

Comment:
Because I noticed that `resolve` just takes a Proc object, I figured I could give it anything as long as it responds to the method `call`. Duck typing for the win in this case, since this is much nicer than `-> (obj, args, ctx) { CreatesDataRequest.call(obj, args, ctx) }`